homeassistant/components/sonos/entity.py

```diff
@@ -57,6 +59,10 @@ async def async_added_to_hass(self) -> None:
             )
         )
 
+    async def async_will_remove_from_hass(self) -> None:
+        """Clean up when entity is removed."""
+        self.hass.data[DATA_SONOS].entity_id_mappings.pop(self.entity_id, None)
```

Comment:
Originally made it `pop` as a defensive measure. I couldn't think of a possible way this could fail, but then I considered modifications to the `entity_id` during runtime. However that ends up working fine as it runs through both `async_will_remove_from_hass` and `async_added_to_hass`. Adjusting with your suggestion. 👍